packages/typespec-azure-resource-manager/lib/arm.foundations.tsp

```diff
@@ -99,26 +99,32 @@ enum ResourceHome {
 @doc("""
 The intended executor of the operation; as in Resource Based Access Control (RBAC) and audit logs UX. Default value is "user,system"
 """)
-enum Origin {
+union Origin {
   @doc("Indicates the operation is initiated by a user.")
-  user,
+  user: "user",
 
   @doc("Indicates the operation is initiated by a system.")
-  system,
+  system: "system",
 
   @doc("Indicates the operation is initiated by a user or system.")
-  `user,system`,
+  `user,system`: "user,system",
+
+  @doc("modelAsString=true")
+  string,
 }
 
 /**
- * Enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs.
+ * Extensible enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs.
  */
 @doc("""
-Enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs.
+Extensible enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs.
 """)
-enum ActionType {
+union ActionType {
   @doc("Actions are for internal-only APIs.")
-  Internal,
+  Internal: "Internal",
+
+  @doc("modelAsString=true")
```

Comment:
Not sure we need this

Reply:
I don't know, this is a doc talking about some autorest property which we should try to detatch arm from autorest emitter so this feels kinda useless, people should get used to extensible unions anyway

Reply:
This could show up in the docs depending on who's emitting it, typescript for example might show a doc here